kotlinpoet/src/test/java/com/squareup/kotlinpoet/ValTypeSpecTest.kt

```diff
@@ -0,0 +1,236 @@
+package com.squareup.kotlinpoet
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.kotlinpoet.KModifier.INLINE
+import com.squareup.kotlinpoet.KModifier.PRIVATE
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class ValTypeSpecTest(private val useVal: Boolean) {
+
+  companion object {
+    @JvmStatic
+    @Parameterized.Parameters(name = "val={0}")
+    fun data(): Collection<Array<Any>> {
+      return listOf(
+        arrayOf(true),
+        arrayOf(false)
+      )
+    }
+  }
+
+  private val modifier = if (useVal) KModifier.VAL else INLINE
+  private val modifierString = modifier.keyword
+
+  private fun classBuilder() = if (useVal) {
```

Comment:
Guess this is more consistent (than `.classBuilder().addModifiers(INLINE)`) with things like `enumBuilder` or `annotationBuilder`.